packages/design-system/src/scalars/components/amount-field/use-amount-field.tsx

```diff
@@ -38,25 +39,56 @@ export const useAmountField = ({
   onChange,
   onBlur,
   tokenIcons,
+  precision,
   viewPrecision,
   trailingZeros,
 }: UseAmountFieldProps) => {
   const currentValue = value ?? defaultValue;
 
   const baseValue =
-    type === "Amount" || type === "AmountPercentage"
-      ? (currentValue as number | undefined)
-      : type === "AmountCurrencyUniversal"
-        ? (currentValue as AmountCurrencyUniversal).amount
-        : (currentValue as AmountCurrencyFiat | AmountCurrencyCrypto).amount;
+    currentValue === undefined
+      ? undefined
+      : type === "Amount" || type === "AmountPercentage"
+        ? (currentValue as number | undefined)
+        : type === "AmountCurrencyUniversal"
+          ? (currentValue as AmountCurrencyUniversal).amount
+          : (currentValue as AmountCurrencyFiat | AmountCurrencyCrypto).amount;
+
+  const valueSelect =
+    currentValue === undefined
+      ? undefined
+      : type === "AmountCurrencyFiat"
+        ? (currentValue as AmountCurrencyFiat).currency
+        : type === "AmountCurrencyCrypto"
+          ? (currentValue as AmountCurrencyCrypto).currency
+          : type === "AmountCurrencyUniversal"
+            ? (currentValue as AmountCurrencyUniversal).currency
+            : undefined;
+
   const isBigInt =
     type === "AmountCurrencyCrypto" ||
     (type === "AmountCurrencyUniversal" &&
       isValidBigInt(baseValue?.toString()));
   const [inputFocused, setInputFocused] = useState(false);
 
-  const formattedAmountState = baseValue?.toString() ?? "";
-  const rawAmountState = baseValue?.toString() ?? "";
+  // // Handle the complete value
+  const [rawAmountState, setRawAmountState] = useState(
+    baseValue?.toString() ?? "",
+  );
+
+  // Handle the formatted value
+  const [formattedAmountState, setFormattedAmountState] = useState(
+    baseValue?.toString() ?? "",
+  );
+  // Funciona para datos compuestos
+  useEffect(() => {
+    setRawAmountState(baseValue === undefined ? "" : baseValue.toString());
+    setFormattedAmountState(
+      baseValue === undefined ? "" : baseValue.toString(),
+    );
+  }, [baseValue, type]);
```

Comment:
why do we need a state that is going to be updated in an effect later? is it not better just create the vars and assign the value?

```
const rawAmountState = baseValue?.toString() ?? "";
const formattedAmountState = baseValue?.toString() ?? "";
```

Reply:
you are right , done